### PR TITLE
breaking: migrate to cjs

### LINKED ISSUE #162: migrate to cjs
We should consider to migrate to cjm.

Right now:

```js
const app = require('fastify')({ logger: true })
app.register(require('fastify-orama'))
app.get('/', async (request, reply) => {
  return { hello: 'world' }
})
app.post('/', async (request, reply) => {
  return request.body
})
app.listen({ port: 8080 })

```

Triggers:

```
/Users/mspigolon/workspace/_experiments/asd/asd.js:2
app.register(require('fastify-orama'))
             ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/mspigolon/workspace/_experiments/asd/node_modules/fastify-orama/index.js from /Users/mspigolon/workspace/_experiments/asd/asd.js not supported.
Instead change the require of index.js in /Users/mspigolon/workspace/_experiments/asd/asd.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/Users/mspigolon/workspace/_experiments/asd/asd.js:2:14) {
  code: 'ERR_REQUIRE_ESM'
}
```

Instead, working with cjs - it works with both cjs and esm without any issues.

### PR DESCRIPTION
**Pull request by Eomm**
fixes #162 

I think after this PR, we could ship the v1

WDYT @mateonunez?

### CODE REVIEW

**Comment by Eomm** [comment]
TS is failing 💩

**Comment by Eomm** [comment]
CC I wrote the TS part with @allevo @lmammino (4 moral support)

**Comment by Puppo** [comment]
@Eomm and @mateonunez, I think there is the same problem with the CI here

**Comment by mateonunez** [comment]
Sigh. I wasn't able to reproduce it locally. We need to fix that flaky before `v1.0`. 

Do you have more information about the error, @Puppo? In that case, we can open an issue with all the needed information to reproduce it locally and provide a fix.

**Comment by Puppo** [comment]
@mateonunez  I don't have any info yet. It seems to be stuck but randomly. I'll look it over.

File: index.d.ts [hunk 1]
```diff
@@ -1,42 +1,47 @@
 import type { FastifyPluginCallback } from 'fastify'
-import type { Document, Orama, ProvidedTypes, Results, SearchParams, create } from '@orama/orama'
+import type { TypedDocument, insert, Orama, Results, SearchParams, create, AnyOrama, PartialSchemaDeep, Schema } from '@orama/orama'
```
Comment (mateonunez) [inline index.d.ts:2]: Can `insert` and `AnyOrama` be removed here? They are no longer used.

**Review by mateonunez — COMMENTED** [review]
The new typing system rocks, man! Thanks a lot.

I think we can send these improvements to `v1.0`, we just need to verify why tests sometimes struggle with Node v20.

**Comment by Puppo** [comment]
> The new typing system rocks, man! Thanks a lot.
> 
> I think we can send these improvements to `v1.0`, we just need to verify why tests sometimes struggle with Node v20.

@mateonunez, the problem with the test is something relative to the reporter; I'm on it to understand the problem; it seems something rests ongoing and doesn't close the process. But it's random and not easy to debug.

**Comment by Puppo** [comment]
I think this is the [issue](https://github.com/nodejs/node/issues/49344)

**Comment by mateonunez** [comment]
> I think this is the [issue](https://github.com/nodejs/node/issues/49344)

Nice catch, @Puppo! 

Any concerns on this, @Eomm? Should we rollback to node-tap in your opinion?

**Comment by Eomm** [comment]
> Should we rollback to node-tap in your opinion?

I would wait a sec for a fix before switching.

Tap changed license and I see everybody is migrating to node test runner.

**Comment by Puppo** [comment]
Any new here guys?
@mateonunez or @Eomm please, run the CI again meanwhile to check everything is okay.
The error is random so if you try twice or three time you should see the 🟢  light.

**Comment by mateonunez** [comment]
> Any new here guys?
> @mateonunez or @Eomm please, run the CI again meanwhile to check everything is okay.
> The error is random so if you try twice or three time you should see the 🟢  light.

I saw a new version of Node has been released, but don't know if it fixes the issue you attached. I'll read the changelog later. Meanwhile I've re-run the CI.

**Comment by Eomm** [comment]
Green

missing approve 😄

**Review by mateonunez — APPROVED** [review]
LGTM

I'll ship the stable version of this plugin tonight.

Thank you so much guys, @Eomm, @Puppo! ♥️

**Pull request merged**